src/ruby/pb/test/client.rb

```diff
@@ -502,6 +514,163 @@ def cancel_after_first_response
     op.wait
   end
 
+  def unimplemented_method
+    begin
+      resp = @stub.unimplemented_call(Empty.new)
+    rescue GRPC::BadStatus => e
+      if e.code != GRPC::Core::StatusCodes::UNIMPLEMENTED
+        fail AssertionError,
+          "Expected status 12 (UNIMPLEMENTED). Received: #{e.code}"
+      end
+    rescue Exception => e
+      fail AssertionError, "Expected BadStatus. Received: #{e.inspect}"
+    end
+  end
+
+  def unimplemented_service
+    begin
+      resp = @stub.unimplemented_call(Empty.new)
+    rescue GRPC::BadStatus => e
+      if e.code != GRPC::Core::StatusCodes::UNIMPLEMENTED
+        fail AssertionError,
+          "Expected status 12 (UNIMPLEMENTED). Received: #{e.code}"
+      end
+    rescue Exception => e
+      fail AssertionError, "Expected BadStatus. Received: #{e.inspect}"
+    end
+  end
+
+  def status_code_and_message
+
+    # Function wide constants.
+    message = "test status method"
+    code = 2
+    status = GRPC::Core::StatusCodes::UNKNOWN
+
+    # Testing with UnaryCall.
+    payload = Payload.new(type: :COMPRESSABLE, body: nulls(1))
+    echo_status = EchoStatus.new(code: code, message: message)
+    req = SimpleRequest.new(response_type: :COMPRESSABLE,
+			    response_size: 1,
+			    payload: payload,
+			    response_status: echo_status)
+    seen_correct_exception = false
+    begin
+      resp = @stub.unary_call(req)
+    rescue GRPC::BadStatus => e
+      if e.code != status
```

Comment:
nit: since the recent change to error classes, I think the `rescue BadStatus` comparison could be replaced with something like `rescue GRPC::Core::StatusCodes::UNKNOWN` https://github.com/grpc/grpc/blob/master/src/ruby/pb/test/client.rb#L489